src/test/java/org/influxdb/InfluxDBLogLevelTest.java

```diff
@@ -0,0 +1,50 @@
+package org.influxdb;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import org.influxdb.dto.Pong;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+/**
+ * Test the InfluxDBimpl log level setting from system property.
+ *
+ * @author hoan.le [at] bonitoo.io
+ *
+ */
+@RunWith(JUnitPlatform.class)
+public class InfluxDBLogLevelTest {
+  @Test
+  public void testLogLevelProperties() {
+    
+    String oldLogLevel = System.getProperty(InfluxDB.LOG_LEVEL_PROP);
+    List<String> logLevels = Arrays.asList(null, "NONE", "BASIC", "HEADERS", "FULL", "abc");
+    logLevels.forEach(logLevel -> {
+      System.out.println("LogLevel = " + logLevel);
+      Optional.ofNullable(logLevel).ifPresent(value -> {
+        System.setProperty(InfluxDB.LOG_LEVEL_PROP, value);
```

Comment:
I don't think it's a good idea to change global configurations (in this case, defining global properties) during the test as this **may** affect other tests if they are running in parallel. Also, if your test fail here for some reason, you will exit the test case (line 38: fail(e)) and **may** leave the system with an incorrect level configuration.
My example may be too much for this simple test but my point here is to make you aware that this is, in general, a bad practice.

Reply:
add try/finally block

For now I see the Surefire plugin config runs tests in sequential. however add NotThreadSafe annotation to make sure Surefire will execute the test sequentially
http://maven.apache.org/surefire/maven-surefire-plugin/examples/fork-options-and-parallel-execution.html